### PR TITLE
New Module Hooks API

### PR DESCRIPTION
**Pull request by filrak**
### Related issues

-
### Short description and why it's useful

This change is meant to prevent future problems with before/after registration hooks in modules. Grouping it into one object allows much easier modifications and future extending than relying on params order. it's the very same approach that nuxt is using with their modules.

TODO:
- [x] Changelog

(describe in a few words what is this Pull Request changing and why it's useful)

### Screenshots of visual changes before/after (if there are any)

-
### Screenshot of passed e2e tests (if you are using our standard setup as a backend)

-

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility)
- [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/Upgrade%20notes.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

### Contribution and currently important rules acceptance

- [ ] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)
- [ ] I read the [TypeScript Action Plan](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/TypeScript%20Action%20Plan.md) and adjusted my PR according to it
- [ ] I read about [Vue Storefront Modules](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md) and I am aware that every new feature should be a module

### CODE REVIEW

**Comment by pkarw** [comment]
It's way better to pass the VSF object because it will allow us to extend `before` and `after` hooks without changing the definitions. 

However, there are currently some modules on Github created with the previous approach so we should be compatible with it (I mean with passing Vue, config .. as separate arguments). Would it be possible?

**Comment by filrak** [comment]
@pkarw i already spoke withe xtension devs and will propose a proper PRs to their repos ;)

File: core/scripts/server.js [hunk 1]
```diff
@@ -144,20 +144,87 @@ app.get('*', (req, res, next) => {
       next()
     }
   }
-
+  const HTMLContent = `<html lang="en">
```
Comment (patzick) [inline core/scripts/server.js]: i think this should be imported from separate file

File: core/modules/cart/hooks/afterRegistration.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import * as types from './../store/mutation-types'
 
-export function afterRegistration(Vue, config, store, isServer) {
+export function afterRegistration({ { Vue, config, store, isServer } }) {
```
Comment (patzick) [inline core/modules/cart/hooks/afterRegistration.ts]: here we have double parenthesis

File: core/scripts/server.js (hunk 1 above)
Comment (filrak) [reply core/scripts/server.js]: yeah, i had no idea it was included here btw, thought i emrged it before

**Pull request merged**